Sources/SymbolKit/UnifiedSymbolGraph/UnifiedSymbolGraph.swift

```diff
@@ -160,6 +160,190 @@ extension UnifiedSymbolGraph {
+        // 1. A symbol is not overloaded in one symbol graph, but is overloaded by a sibling symbol
+        //    in another symbol graph.
+        // 2. A group of overloads contains a different set of symbols between different symbol
+        //    graphs, but the same symbol was chosen to be cloned as the overload group symbol.
+        // 3. A group of overloads contains a different set of symbols between different symbol
+        //    graphs, but different symbols were chosen to be cloned as their respective overload
+        //    group symbols.
+        //
+        // In situation 1, the unified relationships match the most complete picture of all the
+        // overloads (in that there is an overload group that exists and there are relationships
+        // that point to it), but the symbol that is in both graphs only has an OverloadData mixin
+        // connected to the selector for one symbol graph. Depending on how the mixins are loaded
+        // later, this could lead to an inconsistent view where an `overloadOf` relationship exists,
+        // but the client doesn't load the matching OverloadData mixin. In this case, we should
+        // effectively promote the OverloadData to the whole-graph scope.
+        //
+        // In situation 2, we have symbols that look roughly like this:
+        //
+        // ┌───────────────────────────┐  ┌───────────────────────────┐
+        // │ overloadA (1) ─┬─► groupA │  │ overloadA (1) ─┬─► groupA │
+        // │ overloadB (2) ─┤          │  │ overloadC (2) ─┤          │
+        // │ overloadC (3) ─┘          │  │ overloadD (3) ─┘          │
+        // └───────────────────────────┘  └───────────────────────────┘
+        //
+        // The unified relationships are still consistent (there's one overload-group symbol and all
+        // the overloaded symbols point to it), but `overloadC` has different OverloadData depending
+        // on which selector you're using: In one graph it's at index 3, but in the other graph it's
+        // at index 2. In this case we need to recalculate the overload indices based on an ordering
+        // that takes all of the overloaded symbols into account.
+        //
+        // Finally, in situation 3 the relationship graph looks like this:
+        //
+        // ┌───────────────────────────────────────────┐
+        // │                 overloadA (1) ─┬─► groupA │
+        // │ groupB ◄─┬─ (1) overloadB (2) ─┘          │
+        // │          └─ (2) overloadC                 │
+        // └───────────────────────────────────────────┘
+        //
+        // In addition to overloadB having different indices per graph, the unified relationships
+        // are also inconsistent: there are two overload groups that refer to different subsets of
+        // symbols! In this case, we need to recalculate the overload indices like in situation 2
+        // above, but we also need to determine which overload group to keep as the "canonical" one.
```

Comment:
We should not keep the extra overload groups in the unified symbol graph, because they do not reflect the overloaded symbols of the unified graph. For example, if we kept these extra overload groups in the graph, then Swift-DocC would create superfluous overload pages, with duplicate information, that is already reflected in the "canonical" overload group.

This is the primary reason i did this work in SymbolKit rather than in Swift-DocC: If i had to filter these pages after the fact, i would have to either perform this filtering before all the information is available in DocC's data structures, or else violate invariants about data structures that should not be modified to add or remove new nodes after they've been registered. Overload group symbols are by definition "not real symbols": They do not reflect actual API from the source code. They instead reflect a grouping of other "real" symbols that can be used as a convenience for display. If more than one overload group references the same set of symbols, then that creates more of the clutter in an API view that we are trying to clean up with this project in the first place.